src/broker.rs

```diff
@@ -0,0 +1,99 @@
+//!
+//! The broker actor.
+//!
+
+use actix::{Actor, Context, Handler, Message};
+use log::info;
+use osaka::{osaka, Future, FutureResult, Poll};
+
+use carrier::{endpoint::Endpoint, error::Error, identity::Identity};
+
+pub struct BrokerActor {
+    endpoint: Option<Endpoint>,
+}
+
+impl BrokerActor {
+    pub fn new(poll: Poll) -> Self {
+        match Self::endpoint(poll).run() {
+            Ok(endpoint) => {
+                info!("Carrier Endpoint Actor has been started");
+                Self {
+                    endpoint: Some(endpoint),
+                }
+            },
+            Err(error) => panic!(error),
+        }
+    }
+
+    #[osaka]
+    fn endpoint(poll: Poll) -> Result<Endpoint, Error> {
+        let config = carrier::config::load()?;
+
+        let mut endpoint = carrier::endpoint::EndpointBuilder::new(&config)?.connect(poll.clone());
+        let endpoint: Endpoint = osaka::sync!(endpoint)?;
+
+        Ok(endpoint)
+    }
+}
+
+pub enum DeviceStatus {
+    Online,
+    Offline,
+    Unknown,
+}
+
+pub struct BrokerActorMessage(pub Identity);
+
+impl Message for BrokerActorMessage {
+    type Result = Result<DeviceStatus, Error>;
+}
+
+impl Actor for BrokerActor {
+    type Context = Context<Self>;
+}
+
+impl Handler<BrokerActorMessage> for BrokerActor {
+    type Result = Result<DeviceStatus, Error>;
+
+    fn handle(&mut self, msg: BrokerActorMessage, _ctx: &mut Context<Self>) -> Self::Result {
+        let mut endpoint = self.endpoint.take().expect("Endpoint acquisition bug");
+        endpoint.connect(msg.0)?;
+        let (endpoint, result) = outgoing_connect(endpoint).run();

```

Comment:
this is just lazy.
if you're not polling the endpoint, it will eventually time out, but you're not cleaning up.

so if actix remains idle for the timeout period (default 2 minutes),  the next api call will hang forever.

Reply:
I see. Well, I couldn't know that. It could have been improved. Sorry, if it is not what you needed.